src/coreComponents/finiteVolume/TwoPointFluxApproximationWithGraph.cpp

```diff
@@ -0,0 +1,143 @@
+ */
+
+/**
+ * @file TwoPointFluxApproximationWithGraph.cpp
+ *
+ */
+
+#include <unistd.h>
+#include <limits.h>
+
+#include "TwoPointFluxApproximationWithGraph.hpp"
+#include "finiteVolume/FluxApproximationBase.hpp"
+#include "finiteVolume/CellElementStencilTPFA.hpp"
+#include "mesh/FaceElementRegion.hpp"
+#include "meshUtilities/ComputationalGeometry.hpp"
+#include "LvArray/src/tensorOps.hpp"
+#include "mesh/GraphFromText.hpp"
+#include "mesh/GraphEdge.hpp"
+
+
+namespace geosx
+{
+
+using namespace dataRepository;
+
+TwoPointFluxApproximationWithGraph::TwoPointFluxApproximationWithGraph( std::string const & name,
+                                                      Group * const parent )
+  : TwoPointFluxApproximation( name, parent )
+{
+    registerWrapper( viewKeyStruct::graphString, &m_graphString )->
+    setInputFlag( InputFlags::OPTIONAL )->
+    setDescription( "Name of the graph to link to the graph." );
+
+
+}
+
+void TwoPointFluxApproximationWithGraph::computeCellStencil( MeshLevel & mesh) const
+{
+  const Group * tmp = this->GetGroupByPath( m_graphString );
+  GEOSX_ERROR_IF( tmp == nullptr, "Can't find group with path " << m_graphString );
+  const GraphFromText * graph = Group::group_cast< const GraphFromText * >( tmp );
+  ElementRegionManager const & elemManager = *mesh.getElemManager();
+
+  CellElementStencilTPFA & stencil = getStencil< CellElementStencilTPFA >( mesh, viewKeyStruct::cellStencilString );
+
+  
+  std::vector<GraphEdge*> edges = graph->getEdges();
+  std::vector<GraphVertex*> vertices = graph->getVertices();
+  stencil.reserve( edges.size() );
+
+  ElementRegionManager::ElementViewAccessor< arrayView1d< integer const > > const elemGhostRank =
+    elemManager.ConstructArrayViewAccessor< integer, 1 >( ObjectManagerBase::viewKeyStruct::ghostRankString );
+  ElementRegionManager::ElementViewAccessor< arrayView1d< globalIndex const > > const localToGlobalMap =
+    elemManager.ConstructArrayViewAccessor< globalIndex, 1 >( ObjectManagerBase::viewKeyStruct::localToGlobalMapString );
+  /*
+  std::vector <GraphVertex*> new_vertex;
+  for( long unsigned int i=0; i<vertices.size(); i++)
+  {
+    new_vertex.push_back(vertices[i]);
+  }
+  for (long int i = 0; i<elemGhostRank[0][0].size(); i++)
+  {
+    std::cout<<i<<"\n";
+    std::cout<<localToGlobalMap[0][0][i]<<"\n";
+    std::cout<<"\n";
+    for (long unsigned int j = 0; j<new_vertex.size();j++)
+    {
+      if (localToGlobalMap[0][0][i] == new_vertex[i]->getEdgeIndex())
+      {
+        new_vertex.erase(new_vertex.begin()+i);
+      }
+    }
+  }
+  for ( long unsigned int i=0; i<new_vertex.size(); i++)
+  {
+    std::cout<<new_vertex[i]->getEdgeIndex()<<" ";
+    for (long unsigned int j = 0; j<vertices.size();j++)
+    {
+      if (vertices[j]->getEdgeIndex == new_vertex[i]->getEdgeIndex())
+      {
+        graph->RemoveVertex(vertices[j]);
+      }
+    }
+  std::cout<<"\n";
+  }
+  */
+ 
+  for( long unsigned int i=0; i<edges.size(); i++)
```

Comment:
Is there a specific reason not to use one of `localIndex`/`globalIndex`?

Reply:
The only reason is that for the moment I used std::vector instead of the classic GEOSX types. I will modify it to be a localIndex